src/emscripten-optimizer/optimizer-shared.cpp

```diff
@@ -67,7 +56,7 @@ AsmType detectType(Ref node, AsmData *asmData, bool inVarDef, IString minifiedFr
   switch (node[0]->getCString()[0]) {
     case 'n': {
       if (node[0] == NUM) {
-        if (!isInteger(node[1]->getNumber())) return ASM_DOUBLE;
+        if (!wasm::isInteger(node[1]->getNumber())) return ASM_DOUBLE;
```

Comment:
Could we share the support library? Maybe I can make it a git submodule? It's used in a few places, the duplication seems silly.